### PR TITLE
automatically calling tagHandler for tags in the document

### PR DESCRIPTION
**Pull request by phillipskevin**
first piece of https://github.com/canjs/can-view-callbacks/issues/75.

### CODE REVIEW

File: can-view-callbacks.js [hunk 1]
```diff
@@ -6,11 +6,78 @@ var domMutate = require('can-dom-mutate/node');
...
 var makeFrag = require("can-util/dom/frag/frag");
+var canSymbol = require("can-symbol");
 
 //!steal-remove-start
 var requestedAttributes = {};
 //!steal-remove-end
 
+var tags = {};
+
+var GLOBAL = getGlobal();
```
Comment (matthewp) [inline can-view-callbacks.js]: Don't cache this. Even thought this feature will (likely) never run in SSR, having this here might mean someone else uses it in code that will.

File: can-view-callbacks.js [hunk 2]
```diff
@@ -6,11 +6,78 @@ var domMutate = require('can-dom-mutate/node');
...
+var supportsCustomElements = "customElements" in GLOBAL;
+var viewmodelSymbol = canSymbol.for("can.viewModel");
+
+// WeakSet containing elements that have been mounted already
+// and therefore do not need to be mounted again
+var mountedElements = new WeakSet();
+
+var mountNodeAndChildrenIfNecessary = function(node) {
+	var tagName = node.tagName && node.tagName.toLowerCase();
```
Comment (matthewp) [inline can-view-callbacks.js]: nit: `node.localName` is already lower case

File: can-view-callbacks.js [hunk 3]
```diff
@@ -6,11 +6,78 @@ var domMutate = require('can-dom-mutate/node');
...
+					if (!mountedElements.has(addedNode)) {
+						mountNodeAndChildrenIfNecessary(addedNode);
+					}
+				}
+			}
+		}
+	};
+
+	var obs = new MutationObserver(mutationHandler);
+	obs.observe(GLOBAL.document.documentElement, { childList: true, subtree: true });
```
Comment (matthewp) [inline can-view-callbacks.js]: use getGlobal or getDocument

File: can-view-callbacks.js [hunk 4]
```diff
@@ -6,11 +6,78 @@ var domMutate = require('can-dom-mutate/node');
...
+
+	mutationObserverEnabled = true;
+};
+
+var mountExistingElements = function(tagName) {
+	var nodes = GLOBAL.document.getElementsByTagName(tagName);
```
Comment (matthewp) [inline can-view-callbacks.js]: getGlobal or getDocument

**Comment by matthewp** [comment]
I'm a little confused about the flow here. How is anything ever added to `mountedElements`? Only `.has()` is ever used, in which case it will always be empty, no?

**Comment by phillipskevin** [comment]
`mountedElements` will be used by stache. Basically, stache will add elements it has rendered to the `mountedElements` Set so they do not need to be rendered twice.

**Comment by matthewp** [comment]
What does "mounted" mean exactly? Does it mean that the tag handler has been called on it?

**Comment by phillipskevin** [comment]
Yes. If in `mountedElements`, we won't call `tagHandler`.

**Comment by phillipskevin** [comment]
I'm open to ideas for a better name.

**Comment by matthewp** [comment]
So in that case do we need stache to do the job of adding to mountedElements? Can callbacks not do it here? https://github.com/canjs/can-view-callbacks/blob/58fe03530d7bb079b799f7eafb321f352f675b2b/can-view-callbacks.js#L204

**Comment by phillipskevin** [comment]
Yes, you're right we can do it there. I don't think we need the viewModel check either. I'll update the code and tests.

**Comment by matthewp** [comment]
🎉

**Pull request merged**